scipy/stats/morestats.py

```diff
@@ -3478,3 +3478,76 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
         R = np.minimum(1, hypot(sin_mean, cos_mean))
 
     return ((high - low)/2.0/pi) * sqrt(-2*log(R))
+
+
+def quartile_coeff_dispersion(a, q=(0.25, 0.75), axis=0,
+                              interpolation='lower'):
+    """
+    Compute the quartile coefficient of dispersion for a given sample array.
+
+    Given two quartiles Q_low, Q_high from array samples such that
+    Q_low < Q_high, the quartile coefficient of dispersion is calculated by
+    (Q_high - Q_low)/(Q_high + Q_low).
+    by default it is computed using 3rd and 1st quartiles.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.
+    q : array_like of float, optional
+        Sequence of 2 quantiles, such that q[0] < q[1].
```

Comment:
I think the use of `quantiles` and `quartiles` is appropriate, btw. It's unfortunate that the words are so similar, but I think the distinction is made appropriately here.